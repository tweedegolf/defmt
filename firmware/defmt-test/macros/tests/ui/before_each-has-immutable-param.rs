fn main() {}

#[defmt_test_macros::tests]
mod tests {
    #[before_each]
    fn say(name: &str) {
        assert_eq!("name", name);
    }
}
