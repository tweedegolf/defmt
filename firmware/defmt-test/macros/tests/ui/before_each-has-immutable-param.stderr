error: parameter must be a mutable reference (`&mut $Type`)
 --> tests/ui/before_each-has-immutable-param.rs:6:12
  |
6 |     fn say(name: &str) {
  |            ^^^^
